demo/demo.js

```diff
@@ -311,14 +244,29 @@ export function parseRevUrl(url) {
 		placeholder: 'popOut',
 		startAt: 'startAt'
 	};
-	for (let [key, value] of searchParams) {
+	const config = Object.entries(searchParams).reduce((config, [key, value]) => {
 		const configKey = queryConfigMap[key];
 		if (configKey) {
-			result.config[configKey] = value === ''
-				? true
-				: value;
+			config[configKey] = value === '' ? true : value;
 		}
+		return config;
+	}, {});
+
+	result.config = stringifyJson(config);
+	result.embedType = result.videoId ? 'vod' : 'webcast';
+
+	if(result.webcastId) {
```

Comment:
No guarantee that someone will pass a token with the URL. The could embed a non-public webcast or embed a public webcast but have users fill out registration form. Maybe check if the searchParams have a 'token' parameter as well?